src/GridFS/ReadableStream.php

```diff
@@ -252,18 +263,24 @@ private function initBufferFromCurrentChunk(): bool
         }
 
         if (! $this->chunksIterator->valid()) {
-            throw CorruptFileException::missingChunk($this->chunkOffset);
+            $this->chunksIterator = null;
+
+            throw $this->error = CorruptFileException::missingChunk($this->chunkOffset);
```

Comment:
The change in this PR only captures a CorruptFileException. I assume any other exceptions, such as a socket error, would still propagate from the `initBufferFromCurrentChunk()` and `initBufferFromNextChunk()` methods and _not_ be captured.

That seems inconsistent, but before trying to address this I think we should reconsider whether _any_ changes are needed.

Per my [comment in PHPLIB-427](https://jira.mongodb.org/browse/PHPLIB-427?focusedCommentId=6050056&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-6050056), mongodb/mongo-php-library#608 was reported before the GridFS StreamWrapper was changed to throw. The OP encountered an infinite loop precisely because the PHP warning was not interrupting execution (despite being logged).

I expect that [PHPLIB-592](https://jira.mongodb.org/browse/PHPLIB-592) actually would have fixed their issue, and I see no reason to go out of our way to cater to users that may be _ignoring_ exceptions and still interacting with the stream.

If there is still a desire to invalidate the stream after some exception is thrown, I think we'd need some more thorough handling to catch anything that may be potentially thrown -- not just in this class but any of its own method calls. And in that case, preserving the original exception shouldn't be necessary -- just marking that an exception occurred and that future methods should do nothing but report failure seems sufficient. But I'm not convinced that's necessary.

Reply:
Thanks for the analysis. I agree that retriability is incompatible with what I'm proposing here. 

Let's close.